src/osd/OSDMap.cc

```diff
@@ -693,7 +694,19 @@ void OSDMap::Incremental::encode(ceph::buffer::list& bl, uint64_t features) cons
     if (target_v >= 9) {
       encode(new_device_class_flags, bl);
     }
-    ENCODE_FINISH(bl); // osd-only data
+    if (target_v >= 10) {
+      encode(change_stretch_mode, bl);
```

Comment:
Looks to me like the other fields are only interpretted if change_stretch_mode, could we choose not to encode/decode them if !change_stretch_mode?

Reply:
Talked this over and sounds like we don't want to — Sam has visions of switching normal peering to use some of these fields to solve our mis-placement troubles for Pacific.

Reply:
Indeed, at least peering_crush_bucket_barrier (which should be renamed peering_crush_failure_domain) should be totally independent if we wish to make use of it normally.